crates/state/src/account_info.rs

```diff
@@ -32,17 +32,13 @@ impl Default for AccountInfo {
 
 impl PartialEq for AccountInfo {
     fn eq(&self, other: &Self) -> bool {
-        self.balance == other.balance
-            && self.nonce == other.nonce
-            && self.code_hash == other.code_hash
+        (self.balance, self.nonce, self.code_hash) == (other.balance, other.nonce, other.code_hash)
     }
 }
 
 impl Hash for AccountInfo {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.balance.hash(state);
-        self.nonce.hash(state);
-        self.code_hash.hash(state);
+        (&self.balance, self.nonce, &self.code_hash).hash(state)
```

Comment:
this think this now copies the nonce, shouldn't matter for u64 I think but there's no reason to change the previous impl